src/api-service/__app__/timer_workers/__init__.py

```diff
@@ -6,43 +6,44 @@
 import logging
 
 import azure.functions as func
-from onefuzztypes.enums import ScalesetState
+from onefuzztypes.enums import NodeState, PoolState
 
+from ..onefuzzlib.autoscale import autoscale_pool
 from ..onefuzzlib.dashboard import get_event
-from ..onefuzzlib.pools import Scaleset
+from ..onefuzzlib.orm import process_update
+from ..onefuzzlib.pools import Node, Pool, Scaleset
 
 
 def process_scaleset(scaleset: Scaleset) -> None:
     logging.debug("checking scaleset for updates: %s", scaleset.scaleset_id)
 
-    if scaleset.state == ScalesetState.resize:
-        scaleset.resize()
-
     # if the scaleset is touched during cleanup, don't continue to process it
     if scaleset.cleanup_nodes():
         logging.debug("scaleset needed cleanup: %s", scaleset.scaleset_id)
         return
 
-    if (
-        scaleset.state in ScalesetState.needs_work()
-        and scaleset.state != ScalesetState.resize
-    ):
-        logging.info(
-            "exec scaleset state: %s - %s",
-            scaleset.scaleset_id,
-            scaleset.state,
-        )
-
-        if hasattr(scaleset, scaleset.state.name):
-            getattr(scaleset, scaleset.state.name)()
-        return
+    process_update(scaleset)
 
 
 def main(mytimer: func.TimerRequest, dashboard: func.Out[str]) -> None:  # noqa: F841
```

Comment:
This function seems to be doing quite a bit of work. We may want to chop it up more in the future.